src/main/scala/io/iohk/ethereum/network/PeerMessageBusActor.scala

```diff
@@ -0,0 +1,86 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.event.ActorEventBus
+import io.iohk.ethereum.network.p2p.Message
+
+object PeerMessageBusActor {
+  def props: Props = Props(new PeerMessageBusActor)
+
+  sealed trait PeerSelector {
+    def contains(peerId: PeerId): Boolean
+  }
+
+  object PeerSelector {
+
+    case object AllPeers extends PeerSelector {
+      override def contains(p: PeerId): Boolean = true
+    }
+
+    case class WithId(peerId: PeerId) extends PeerSelector {
+      override def contains(p: PeerId): Boolean = p == peerId
+    }
+
+  }
+
+  case class MessageClassifier(messageCodes: Set[Int], peerSelector: PeerSelector)
+
+  case class MessageFromPeer(message: Message, peerId: PeerId)
+
+  case class Subscription(subscriber: ActorRef, classifier: MessageClassifier)
+
+  class PeerMessageBus extends ActorEventBus {
+
+    override type Event = MessageFromPeer
+    override type Classifier = MessageClassifier
+
+    private var subscriptions: Seq[Subscription] = Nil
+
+    override def subscribe(subscriber: ActorRef, to: Classifier): Boolean = {
+      val subscription = Subscription(subscriber, to)
+      if (subscriptions.contains(subscription)) {
+        false
+      } else {
+        subscriptions = subscriptions :+ subscription
+        true
+      }
+    }
+
+    override def unsubscribe(subscriber: ActorRef, from: Classifier): Boolean = {
+      val subscription = Subscription(subscriber, from)
+      if (subscriptions.contains(subscription)) {
+        subscriptions = subscriptions.filterNot(_ == subscription)
+        true
+      } else {
+        false
+      }
+    }
+
+    override def unsubscribe(subscriber: ActorRef): Unit = {
```

Comment:
I'm wondering, should we have a separate `Unsubscribe` message for this function, so that we are able to call it as with `override def unsubscribe(subscriber: ActorRef, from: Classifier): Boolean`? We could also consider adding it later if necessary.